Add file logging during an import using 3rd party lib

ArthurHoaro opened the pull request:
* Logs are stored in `<given folder>/import.log`
  * Basic log in INFO mode (default)
  * Detailled logs in DEBUG mode
  * Unit tested

Choice of library: most stared generic logger in packagist, no extra dependency, PHP compatibility

Example INFO log:

```
[2017-01-27 20:34:35.049333] [info] Starting to parse tests/input/netscape_basic.htm
[2017-01-27 20:34:35.049418] [info] PARSING LINE #0
[2017-01-27 20:34:35.049453] [info] PARSING LINE #1
[2017-01-27 20:34:35.049480] [info] PARSING LINE #2
[2017-01-27 20:34:35.049562] [info] PARSING LINE #3
[2017-01-27 20:34:35.049592] [info] PARSING LINE #4
[2017-01-27 20:34:35.049664] [info] PARSING LINE #5
[2017-01-27 20:34:35.049696] [info] File parsing ended
```

Example DEBUG log

```
[2017-01-27 20:37:07.698795] [info] Starting to parse tests/input/shaarli.htm
[2017-01-27 20:37:07.698930] [info] PARSING LINE #0
[2017-01-27 20:37:07.698966] [debug] [#0] Content: <DL>
[2017-01-27 20:37:07.698999] [info] PARSING LINE #1
[2017-01-27 20:37:07.699028] [debug] [#1] Content: <DT>
[2017-01-27 20:37:07.699059] [info] PARSING LINE #2
[2017-01-27 20:37:07.699089] [debug] [#2] Content: <A HREF="?lY47tw" ADD_DATE="1459371397" PRIVATE="0" TAGS="breton,recipe,kouign,amann,traditional,caramel,butter">Kouign'amann recipes</A><DD>&quot;Is there anything more fabulous than something created through the wonder and miracle of caramelization?&quot;<br><br>- http://www.davidlebovitz.com/2005/08/long-live-the-k/<br>- http://www.bonappetit.com/recipe/kouign-amann<br><br>&quot;It is strictly forbidden to think about diet while you're making a Kouign Amann&quot;
[2017-01-27 20:37:07.699122] [debug] [#2] Link found
[2017-01-27 20:37:07.699155] [debug] [#2] URL found: ?lY47tw
[2017-01-27 20:37:07.699189] [debug] [#2] Title found: Kouign'amann recipes
[2017-01-27 20:37:07.699227] [debug] [#2] Content found: &quot;Is there anyth...
[2017-01-27 20:37:07.699264] [debug] [#2] Tag list: breton recipe kouign amann traditional caramel butter
[2017-01-27 20:37:07.699311] [debug] [#2] Date: 1459371397
[2017-01-27 20:37:07.699347] [debug] [#2] Visibility: public
```

ArthurHoaro committed:
Add file logging during an import using 3rd party lib

  * Basic log in INFO mode (default)
  * Detailled logs in DEBUG mode
  * Unit tested

Choice of library: most stared generic logger in packagist, no extra dependency, PHP compatibility

Also: added a namespace for AutoLoading

nodiscc commented on tests/ImportLoggerTest.php:
```diff
@@ -0,0 +1,72 @@
+    }
+
+    /**
+     * Make sure that the log is created an contains INFO logs, without DEBUG logs.
```
I think this comment line is wrong. Should be

> an contains INFO logs AND DEBUG logs

ArthurHoaro replied:
Thanks! Updated.

virtualtam commented on NetscapeBookmarkParser.php:
```diff
@@ -37,6 +51,15 @@ public function __construct(
+            $logsDir,
+            $logLevel,
+            array(
+                'filename' => 'import.log',
```
- could be parameterized if we want to append logs to an existing file
- the default filename should include the current timestamp to avoid being overriden and keep one log file per imported bookmark file

ArthurHoaro replied:
You're right. Actually I don't know why I used this way, since the logger is PSR compliant, I'll just add a setter.

ArthurHoaro commented:
Updated! As I said, I added a setter to support any logger lib which follows PSR-3 standard, and I also added a namespace to each class for autoloading.

virtualtam reviewed (approved):
Wow, there's even a [`LoggerAwareInterface`](http://www.php-fig.org/psr/psr-3/#psrlogloggerawareinterface)... didn't know JCVD was into web specifications these days :p

ArthurHoaro committed:
NetscapeBookmarkParser is now LoggerAware!

« L'être humain en général dans la vie *react* ; on react ; c'est à dire qu'on fait ce qu'on est supposé faire. Travailler, manger, je m'excuse l'expression... « chier », mais je sais qu'un être humain doit créyer. Y'a des gens qui n'ont pas réussi parce qu'ils ne sont pas **aware**, ils ne sont pas au courant, ils ne sont pas à l'attention de savoir qu'ils existent. Les pauvres, ils savent pas ! (...) Tu veux un mouchoir ? Y'a des gens comme ça, qui ne sont pas aware ; moi je suis aware, tu vois ! C'est un example, je suis aware ! »

virtualtam commented:
_JCVD, about examining the log stack:_

![jcvd-casse-des-briques](https://i.chzbgr.com/full/7153366528/hEB66FEF5/)
